File: lab02/Shape.cpp
// File: Shape.cpp
// Author: Dr. Tia Watts
// Contents: This file contains the implementation of the virtual Shape Class

#include "Shape.h"

Shape::Shape()
{
    side = 0.0;
}

Shape::Shape(const Shape &S)
{
    side = S.side;
}

Shape::~Shape()
{
}

Shape & Shape::operator=(const Shape &S)
{
    if (this == &S)
        return *this;

    side = S.side;
    return *this;
}

float Shape::Area() const
{
    return 0;
}

float Shape::Perimeter() const
{
    return 0;
}

void Shape::WriteArea(ostream &outs) const
{
}

ostream & operator << (ostream &outs, const Shape &S) 
{
    S.WriteArea(outs);
    return outs;
}

File: lab08/Shape.java
// File: Shape.java
// Author: Dr. Watts
// Contents: This file contains the description and implementation
// of a virtual class called Shape. 

import java.text.NumberFormat;
import java.awt.*;

public class Shape
{
    protected int centerX;
    protected int centerY;
	protected int side;

	public Shape ()
	{
		side = 0;
	}
	
	public Shape (Shape S)
	{
		side = S.side;
	}
	
	public double area ()
	{
		return 0;
	}
	
	public double perimeter ()
	{
		return 0;
	}
	
	public String sides ()
	{
		return "0";
	}

    public void paintComponent(Graphics2D g2) {}
	
	public void write ()
	{
   		NumberFormat nf = NumberFormat.getInstance();
   		nf.setMinimumFractionDigits(2);
   		nf.setMaximumFractionDigits(2);
		System.out.print (sides () + " sided shape; ");
		System.out.print (" with perimeter = " + nf.format(perimeter ()));
		System.out.print (" and area = " + nf.format(area ()));
	}

	public String getName ()
	{
		return "Shape";
	}

	public static void main (String [] args)
	{
		Shape S = new Shape ();
		S.write ();
		System.out.println ();
	}
}
